foundations/foundation-vertx/src/main/java/org/apache/servicecomb/foundation/vertx/metrics/DefaultHttpClientMetrics.java

```diff
@@ -133,7 +139,24 @@ public void disconnected(Object webSocketMetric) {
 
   @Override
   public DefaultHttpSocketMetric connected(SocketAddress remoteAddress, String remoteName) {
-    return new DefaultHttpSocketMetric(null);
+    // when host of createEndpoint is not ip but a hostName
+    // get from remoteAddress will return null
+    // in this time need to try again with remoteName
+    // connected is a low frequency method, this try logic will not cause performance problem
+
+    DefaultClientEndpointMetric clientEndpointMetric = this.clientEndpointMetricManager
+        .getClientEndpointMetric(remoteAddress);
+    if (clientEndpointMetric == null) {
+      LOGGER.warn("can not find endpointMetric directly by remoteAddress {}, try again with remoteName {}",
```

Comment:
When will this happen? If it's too frequently, we can add only debug log.

Reply:
I'm not sure about it.  maybe debug is better

Reply:
line from 142 to 145 explain when will get a null result.
but i think the warn log is not necessary, it's a normal status, no need to warn

Reply:
ok, get